Source/WebCore/accessibility/AXObjectCache.cpp

```diff
@@ -2067,7 +2067,48 @@ void AXObjectCache::onStyleChange(Element& element, OptionSet<Style::Change> cha
 #endif // ENABLE(ACCESSIBILITY_ISOLATED_TREE)
 }
 
+HashMap<AXID, LineRange> AXObjectCache::mostRecentlyPaintedText()
+{
+    HashMap<AXID, LineRange> recentlyPaintedText;
+    for (auto renderTextToLineRange : m_mostRecentlyPaintedText) {
+        if (auto* axObject = getOrCreate(renderTextToLineRange.key))
+            recentlyPaintedText.add(axObject->objectID(), renderTextToLineRange.value);
+    }
+    return recentlyPaintedText;
+}
+
 #if ENABLE(ACCESSIBILITY_ISOLATED_TREE)
+void AXObjectCache::onAccessibilityPaintStarted()
+{
+    m_mostRecentlyPaintedText.clear();
+}
+
+void AXObjectCache::onAccessibilityPaintFinished()
+{
+    RefPtr tree = AXIsolatedTree::treeForPageID(m_pageID);
+    if (!tree)
+        return;
+
+    for (auto iterator = m_mostRecentlyPaintedText.begin(); iterator != m_mostRecentlyPaintedText.end(); ++iterator) {
+        const auto& renderText = iterator->key;
+        // FIXME: Use InlineIteratorLogicalOrderTraversal instead. Otherwise we'll do the wrong thing for mixed direction
```

Comment:
Is there a radar for this?

Reply:
I just filed:

https://bugs.webkit.org/show_bug.cgi?id=294632

And added a link to this bug in code comments in both places that will need changes.